### PR TITLE
Upgrade to toml v0.5.11

### LINKED ISSUE #1894: TOML parse error: Invalid table header: Duplicate key `version`
Run `ruff .` with this `pyproject.toml` in an empty directory:

```toml
# pyproject.toml

[tool.hatch]
version.source = "vcs"

[tool.hatch.version.raw-options]
local_scheme = "no-local-version"
```

Outputs error only without `--isolated` in newest 0.0.222:

```console
$ ruff --version
ruff 0.0.222
$  ruff .
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`

$ ruff . --isolated
warning: No Python files found under the given path(s)
```

This file is parsed fine by other tools, such as Hatchling, python-build, Black, isort etc. Fuller version at https://github.com/hugovk/norwegianblue/

Error both with and without `--isolated` in 0.0.219:

```console
$ ruff --version
ruff 0.0.219
$ ruff .
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`

$ ruff . --isolated
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`
```

Error only without `--isolated` in 0.0.220:

```console
$ ruff --version
ruff 0.0.220
$ ruff .
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`

$ ruff . --isolated
warning: No Python files found under the given path(s)
```

### PR DESCRIPTION
**Pull request by charliermarsh**
In #1680, we moved over to `toml_edit`. But it looks like `toml` now uses `toml_edit`, and has implemented some improvements (e.g., this closes #1894).

### CODE REVIEW

**Comment by charliermarsh** [comment]
\cc @messense

**Comment by messense** [comment]
Looks like the released version on crates.io doesn’t use toml_edit currently: https://crates.io/crates/toml/0.5.11/dependencies

**Comment by charliermarsh** [comment]
Oh, strange. Any idea what the state of the crate is? Should I revert this?

**Comment by messense** [comment]
Only `toml` released a new version without noticeable any behavior changes, `toml_edit` is still on 0.17.1 which doesn't resolve the issue.

> Should I revert this?

For now you can revert it to be more compliant with TOML 1.0. At least a new `toml_edit` release a required for #1894.

Or we can just depends on `toml` main branch using git dependency?

**Comment by charliermarsh** [comment]
Interesting, I did run through the example described in that issue and it seemed to work with the new crate. Is that unexpected?

**Comment by charliermarsh** [comment]
@messense - It looks like v0.5.11 was mostly about deprecations in advance of the toml-edit migration. I am confused though why this now works:

```toml
[tool.hatch]
version.source = "vcs"

[tool.hatch.version.raw-options]
local_scheme = "no-local-version"
```

So, I might leave it as-is for now? Is that bad? What did we lose by not supporting TOML 1.0?

**Comment by messense** [comment]
It still fails for me?

```
cargo run
   Compiling toml-test v0.1.0 (/private/tmp/toml-test)
    Finished dev [unoptimized + debuginfo] target(s) in 0.16s
     Running `target/debug/toml-test`
Err(
    Error {
        inner: ErrorInner {
            kind: Custom,
            line: Some(
                5,
            ),
            col: 0,
            at: Some(
                55,
            ),
            message: "duplicate key: `version`",
            key: [
                "tool",
                "hatch",
            ],
        },
    },
)
```

**Cargo.toml**

```toml
[package]
name = "toml-test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
toml = "0.5.11"
```

**src/main.rs**

```rust
fn main() {
    let content = r#"# pyproject.toml

[tool.hatch]
version.source = "vcs"

[tool.hatch.version.raw-options]
local_scheme = "no-local-version""#;
    println!("{:#?}", toml::from_str::<toml::Value>(content));
}
```

**Comment by charliermarsh** [comment]
Interesting, let me try again in the morning.

**Comment by charliermarsh** [comment]
Yeah I can reproduce that, but interestingly, this works fine:

```toml
let content = r#"# pyproject.toml

[tool.hatch]
version.source = "vcs"

[tool.hatch.version.raw-options]
local_scheme = "no-local-version""#;

println!("{:#?}", toml::from_str::<Pyproject>(content));
```
